### PR TITLE
Add flexGrow prop to FlexItem

### PR DESCRIPTION
**Pull request by lzhang-cbi**
## Packages changed

- [ ] `fds-dictionary`
- [ ] `fds-styles`
- [x] `fds-components`
- [ ] `fds-mui-theme`
- [ ] `fds-icons`

## Checklist
- [ ] Docs have been rebuilt (`yarn build:full`)
- [ ] All unit tests pass
- [ ] Snapshots have been updated
- [ ] No lint errors or warnings have been introduced
- [ ] **[Version bumped](https://github.com/cbinsights/form-design-system#updating-version-number) if appropriate**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -7,7 +7,7 @@
 * Jenkins populates the patch version depending on the branch.
 */
 
-String VERSION = "2.3"
+String VERSION = "2.3.1"
```
Comment (akdetrick) [inline Jenkinsfile]: The patch version is set by Jenkins, so this should get a minor version bump instead:

```
String VERSION = "2.4"
```

**Comment by akdetrick** [comment]
You probably also need to run `yarn test -u` from repo root to update snapshots

File: Jenkinsfile (hunk 1 above)
Comment (lzhang-cbi) [reply Jenkinsfile]: im getting error on running yarn test -u, parameter u not found

Comment (akdetrick) [reply Jenkinsfile]: I'll look into that; looks like you don't need to update any snapshots for this change anyway

Comment (lzhang-cbi) [reply Jenkinsfile]: cool!

**Pull request merged**